src/OneDriveSdk/Authentication/AuthenticationProvider.cs

```diff
@@ -151,6 +150,66 @@ protected AccountSession GetAuthenticationResultFromCache()
             return null;
         }
 
+        internal async Task<string> GetAuthorizationCodeAsync(string returnUrl = null)
+        {
+            if (this.ServiceInfo.WebAuthenticationUi != null)
+            {
+                returnUrl = returnUrl ?? this.ServiceInfo.ReturnUrl;
+
+                var requestUriStringBuilder = new StringBuilder();
+                requestUriStringBuilder.Append(this.ServiceInfo.AuthenticationServiceUrl);
+                requestUriStringBuilder.AppendFormat("?{0}={1}", Constants.Authentication.RedirectUriKeyName, returnUrl);
+                requestUriStringBuilder.AppendFormat("&{0}={1}", Constants.Authentication.ClientIdKeyName, this.ServiceInfo.AppId);
+                requestUriStringBuilder.AppendFormat("&{0}={1}", Constants.Authentication.ScopeKeyName, string.Join("%20", this.ServiceInfo.Scopes));
```

Comment:
Looks like you are using a url encoder in the next function, could we use that over here too, or unify how scopes are processed?